src/openms/source/ANALYSIS/QUANTITATION/IsotopeLabelingMDVs.cpp

```diff
@@ -0,0 +1,342 @@
+      experiment_data_peak                  = experiment_data[experiment_data_peak_idx];
+      
+      if ( experiment_data_peak_idx >= 1 && experiment_data_peak != 0.0)
+      {
+        double previous_experiment_data_peak  = experiment_data[experiment_data_peak_idx - 1];
+        double isotopic_purity                = experiment_data_peak_idx / (experiment_data_peak_idx + ( previous_experiment_data_peak / experiment_data_peak));
+        featuremap_with_isotopic_purity.setMetaValue(isotopic_purity_name, isotopic_purity);
+      }
+    }
+  }
+
+  void IsotopeLabelingMDVs::calculateIsotopicPurities(
+    FeatureMap& normalized_featureMap,
+    FeatureMap& featureMap_with_isotopic_purity,
+    std::vector<double>& experiment_data,
+    std::string& isotopic_purity_name)
+  { 
+    for (FeatureMap::Iterator feature_it = normalized_featureMap.begin(); feature_it != normalized_featureMap.end(); feature_it++){
+      Feature feature_with_isotopic_purity;
+      calculateIsotopicPurity( *feature_it, feature_with_isotopic_purity, experiment_data, isotopic_purity_name);
+      featureMap_with_isotopic_purity.push_back(feature_with_isotopic_purity);
+    }
+  }
+
+  
+  void IsotopeLabelingMDVs::calculateMDVAccuracy(
+    Feature& normalized_feature,
+    Feature& feature_with_accuracy_info,
+    std::vector<double>& fragment_isotopomer_measured,
+    std::vector<double>& fragment_isotopomer_theoretical)
+  {
+    feature_with_accuracy_info = normalized_feature;
+    
+    std::vector<double> fragment_isotopomer_abs_diff;
+    for (size_t i = 0; i < fragment_isotopomer_theoretical.size(); ++i) {
+      fragment_isotopomer_abs_diff.push_back(std::abs(fragment_isotopomer_theoretical[i] - fragment_isotopomer_measured[i]));
+    }
+    
+    double diff_mean = 0.0;
+    for (size_t i = 0; i < fragment_isotopomer_abs_diff.size(); ++i) {
+      diff_mean += fragment_isotopomer_abs_diff.at(i);
+    }
+    
+    diff_mean /= fragment_isotopomer_abs_diff.size();
+    
+    std::vector<double> fragment_isotopomer_abs_diff_;
+    for (size_t i = 0; i < fragment_isotopomer_abs_diff.size(); ++i) {
+      fragment_isotopomer_abs_diff_.push_back(std::abs(fragment_isotopomer_abs_diff[i] - diff_mean));
+    }
+    
+    diff_mean = 0.0;
+    for (size_t i = 0; i < fragment_isotopomer_abs_diff_.size(); ++i) {
+      diff_mean += fragment_isotopomer_abs_diff_.at(i);
+    }
+    
+    diff_mean /= fragment_isotopomer_abs_diff_.size();
+    
+    feature_with_accuracy_info = normalized_feature;
+    feature_with_accuracy_info.setMetaValue("average_accuracy", diff_mean);
+  }
+
+
+  void IsotopeLabelingMDVs::calculateMDVAccuracies(
+    FeatureMap& normalized_featureMap,
+    FeatureMap& featureMap_with_accuracy_info,
+    std::vector<double>& fragment_isotopomer_measured,
+    std::vector<double>& fragment_isotopomer_theoretical)
+  {
+    for (FeatureMap::Iterator feature_it = normalized_featureMap.begin(); feature_it != normalized_featureMap.end(); feature_it++){
```

Comment:
I don't know if applicable here but if possible we prefer range loops
```
for (const Feature&  feature : normalized_featureMap)
```

Reply:
please check also other places in code

Reply:
done